sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -556,10 +581,233 @@ public String toJson() {
                 return snapshot;
             }
 
+            this.systemUsage = ClientSideRequestStatistics.fetchSystemInformation();
             return this.cachedRequestDiagnostics = getRequestDiagnostics();
         }
     }
 
+    /**
+     * Gets the UserAgent header value used by the client issueing this operation
+     * NOTE: this information is not included in the json representation returned from {@link #toJson()} because it
+     * is usually only relevant when thresholds are violated, in which case the entire diagnostics json-string is
+     * included. Calling this method will lazily collect the user agent - which can be useful when writing
+     * a custom {@link CosmosDiagnosticsHandler}
+     * @return the UserAgent header value used for the client that issued this operation
+     */
+    public String getUserAgent() {
+        return this.userAgent;
+    }
+
+    /**
+     * Returns the connection mode used in the client.
+     * NOTE: this information is not included in the json representation returned from {@link #toJson()} because it
+     * is usually only relevant when thresholds are violated, in which case the entire diagnostics json-string is
+     * included. Calling this method will lazily collect the user agent - which can be useful when writing
+     * a custom {@link CosmosDiagnosticsHandler}
+     * @return the connection mode used in the client.
+     */
+    public String getConnectionMode() {
+        return this.connectionMode;
+    }
+
+    private static void addRequestInfoForGatewayStatistics(
+        ClientSideRequestStatistics requestStats,
```

Comment:
nit: probably just ActivityId? or remove the the gatewayStats?

Reply:
Fixed in next iteration